### ISSUE TITLE
Viewing and retrieving data

### ISSUE THREAD

**Issue by ChrisAmbrose-UofS** [NicoKiaru/LimeSeg#9]
Hi Nicolas,
I am contacting you here because unfortunately your institution is blocking my emails unless I somehow get authorized. I have a few LimeSeg questions, hoping you can help!

1. Is there any way to determine which cell number corresponds to the cell in the 3D display? For example, I perform the segregation and it finds fifteen cells, listed in the 'cells explorer' table. Currently the only way I can figure out how to determine which cell is which is by selecting a cell from the table, then going to the GUI and changing some aspect of that cell (such as color or display mode, etc).

2. When re-loading a dataset (using 'loadStateFromXmlPly'), there are some issues with positioning and zoom in the 3D display window: 
2a. The cells are not centered anymore. The upper left corner appears to be the new center, which corresponds to x=0, y=0, z=0. Changing the 3D view center seems arbitrary and depends on the dataset, so finding the true center for correct rotations is trial and error. 
2b. Similarly, for the "set3DViewZoom", entering a value of 1 (or 1.0) zooms up so high you can only see a small part of the data. Values less than 1 seem to be ignored. Entering zero results in the dataset being nowhere to be found in the 3D display. The reason I ask these two questions is for accurately scaling images for using in my publication figures. I currently 'eyeball it' and take a screenshot.

3. Is there any way to obtain the "results" window that displays the numerical data table, other than the auto-generated one that appears following completion of a segregation?

Thank you for any help!
Chris

**Comment by ChrisAmbrose-UofS** [comment]
Update to question 2a:  I figured out to just check the xy size of the original image and enter that into the set3DViewCenter. Works perfectly.

**Comment by NicoKiaru** [comment]
Hello Chris,

Thanks for keeping up with LimeSeg. Actually you can also post on the imagej forum in case (forum.image.sc) and put a `LimeSeg` tag. And probably you were contacting me at unige.ch, but I've changed position recently and I can be contacted at epfl.ch now.

> 1. Is there any way to determine which cell number corresponds to the cell in the 3D display?

Before, not really, but now there is. If you update FIJI and LimeSeg (new version as of yesterday), you now have these options, which can be combined, all accessible from LimeSeg GUI (and also scripting):
* 3D View > `enableTrackCurrentCell` -> this centers the 3D view on the current selected cell.
* 3D View > `disableTrackCurrentCell` -> disable the centering
* 3D View > `setCurrentCellColorLUT` -> displays the current cell in green, others in red
* 3D View > `setDefaultColor3DDisplay` -> restores standard color display
* Current Cell > `selectNextCell` -> select the next cell as currentcell in allcells array
* Current Cell > `selectPreviousCell` -> select the previous cell as currentcell in allcells array
* Double clicking on the Cells explorer tab also works

> 2b. Similarly, for the "set3DViewZoom", entering a value of 1 (or 1.0) zooms up so high you can only see a small part of the data. Values less than 1 seem to be ignored. 

Hum, this should work. Try very little values like '0.002' and go by powers of ten. Let me know if this still doesn't work. Maybe try with comma and checks that there is an error in the console window.

> 3. Is there any way to obtain the "results" window that displays the numerical data table, other than the auto-generated one that appears following completion of a segregation?

Try this groovy script : you can arrange it as you like

```
#@Integer frameNumber
#@Boolean meshExists
#@Double realXYPixelSize

import eu.kiaru.limeseg.LimeSeg;
// Objects used by LimeSeg
import eu.kiaru.limeseg.struct.Cell;
import eu.kiaru.limeseg.struct.CellT;
import eu.kiaru.limeseg.struct.DotN;
import eu.kiaru.limeseg.struct.Vector3D;


for (Cell c:LimeSeg.allCells) {
	LimeSeg.currentCell=c;
	CellT ct = c.getCellTAt(frameNumber) // Get the Cell at first timepoint
	if (ct!=null) {
		ct.updateCenter();
   		println("Cell Name:"+ct.c.id_Cell);
   			println("Number of Surfels=\t"+ ct.dots.size());
   			println("Center X=\t"+ ct.center.x);
   			println("Center Y=\t"+ ct.center.y);
   			println("Center Z=\t"+ (ct.center.z/LimeSeg.opt.getOptParam("ZScale"))+1); // IJ1 style numerotation
   			println("Frame=\t"+ ct.frame);
   			println("Channel=\t"+ ct.c.cellChannel);       			
   			println("Mesh ?=\t"+ (meshExists)?"YES":"NO");
   			if (meshExists) {
   				println("Euler characteristic=\t"+ (ct.dots.size()-3.0/2.0*ct.triangles.size()+ct.triangles.size()));
   				println("Free edges=\t"+ ct.freeEdges);
   				println("Surface=\t"+ ct.getSurface());
   				println("Volume=\t"+ ct.getVolume());
   				println("Real Surface=\t"+ ct.getSurface()*(realXYPixelSize*realXYPixelSize));
   				println("Real Volume=\t"+ ct.getVolume()*(realXYPixelSize*realXYPixelSize*realXYPixelSize));
   			}
	}
	
}
```

**Comment by ChrisAmbrose-UofS** [comment]
Hi Nicolas, 

This is amazing, thank you! I've tried everything out and it works perfectly. I attached two pictures I wanted to show you of images I made with limeseg a while back. The curvature images are exactly what I was hoping to get. They are leaf epidermal cells from Arabidopsis.

Best wishes, and thanks again!
Chris
![Limeseg3D_UnknownNameEpidermis 003](https://user-images.githubusercontent.com/47755821/57737639-9aaa3380-7669-11e9-926f-70971ae7a93d.jpg)
![Limeseg3D_UnknownNameEpidermis 001](https://user-images.githubusercontent.com/47755821/57737642-9da52400-7669-11e9-9054-2716d2fb0589.jpg)
